File: app/src/main/java/com/zzl/top/common/http/IHttpClient.java
package com.zzl.top.common.http;

/**
 * Created by zhangzl
 * time: 2017/10/16 20:40.
 * 描述：抽象接口
 */

public interface IHttpClient {
    IResponse get(IRequest request);

    IResponse post(IRequest request);

    IResponse postJson(IRequest request);

    IResponse get(IRequest request, boolean forceCache);

    IResponse post(IRequest request, boolean forceCache);

    IResponse postJson(IRequest request, boolean forceCache);
}

File: http/src/main/kotlin/com/zzl/net/IHttpClient.kt
package com.zzl.net

/**
 * Created by zhangzl
 * 时间： 2017/10/29 14:07
 * 描述：
 */
interface IHttpClient {
    open fun get(request: IRequest): IResponse

    open fun post(request: IRequest): IResponse

    open fun postJson(request: IRequest): IResponse
}